### DISCUSSION TITLE
Verifast enforces underflow/overflow checks on unsigned integers.

### DISCUSSION THREAD

**Discussion by isubasinghe** [verifast/verifast#499] [Q&A] [answered]
Arithmetic on unsigned integers is defined for all cases in C as far as I am aware. 

I am writing a queue which relies on overflows to wrap around. Would it be possible for me to let verifast allow this overflow to happen on that specific line of code ? 

Alternatively, can I disable overflows/underflows checking for unsigned integers?

**Comment by btj** [comment, marked as answer]
If wraparound semantics (a.k.a. truncating semantics) is intended for a given arithmetic operation, write it as follows: `/*@ truncating @*/ (x + y)`.

**Reply by isubasinghe** [reply]
Thanks a lot!
